### DISCUSSION TITLE
Unable to activate OSD: Failed to connect to bus: No such file or directory

### DISCUSSION THREAD

**Discussion by osaffer** [rook/rook#8875] [General]
Hi,

All my OSD are down. Before upgrading to 1.7.4, they  were up

When I try to start OSD manually, I have the following.

rook-ceph-osd-0-77f5756d7d-bnnsq bootstrap-osd]# ceph-volume lvm activate --all
--> Activating OSD ID 7 FSID 8349ef0c-545e-4f05-8119-14bb153a3311
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-7
Running command: /usr/bin/ceph-bluestore-tool --cluster=ceph prime-osd-dir --dev /dev/ceph-0c3476af-5cb0-4103-a041-0216210761c1/osd-data-6287f955-2cf4-48fa-b044-dcac57a59df0 --path /var/lib/ceph/osd/ceph-7 --no-mon-config
Running command: /usr/bin/ln -snf /dev/ceph-0c3476af-5cb0-4103-a041-0216210761c1/osd-data-6287f955-2cf4-48fa-b044-dcac57a59df0 /var/lib/ceph/osd/ceph-7/block
Running command: /usr/bin/chown -h ceph:ceph /var/lib/ceph/osd/ceph-7/block
Running command: /usr/bin/chown -R ceph:ceph /dev/dm-0
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-7
Running command: /usr/bin/systemctl enable ceph-volume@lvm-7-8349ef0c-545e-4f05-8119-14bb153a3311
 stderr: Failed to connect to bus: No such file or directory
Traceback (most recent call last):
  File "/usr/sbin/ceph-volume", line 11, in <module>
    load_entry_point('ceph-volume==1.0.0', 'console_scripts', 'ceph-volume')()
  File "/usr/lib/python3.6/site-packages/ceph_volume/main.py", line 40, in __init__
    self.main(self.argv)
  File "/usr/lib/python3.6/site-packages/ceph_volume/decorators.py", line 59, in newfunc
    return f(*a, **kw)
  File "/usr/lib/python3.6/site-packages/ceph_volume/main.py", line 152, in main
    terminal.dispatch(self.mapper, subcommand_args)
  File "/usr/lib/python3.6/site-packages/ceph_volume/terminal.py", line 194, in dispatch
    instance.main()
  File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/main.py", line 46, in main
    terminal.dispatch(self.mapper, self.argv)
  File "/usr/lib/python3.6/site-packages/ceph_volume/terminal.py", line 194, in dispatch
    instance.main()
  File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/activate.py", line 373, in main
    self.activate_all(args)
  File "/usr/lib/python3.6/site-packages/ceph_volume/decorators.py", line 16, in is_root
    return func(*a, **kw)
  File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/activate.py", line 254, in activate_all
    self.activate(args, osd_id=osd_id, osd_fsid=osd_fsid)
  File "/usr/lib/python3.6/site-packages/ceph_volume/decorators.py", line 16, in is_root
    return func(*a, **kw)
  File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/activate.py", line 299, in activate
    activate_bluestore(lvs, args.no_systemd)
  File "/usr/lib/python3.6/site-packages/ceph_volume/devices/lvm/activate.py", line 214, in activate_bluestore
    systemctl.enable_volume(osd_id, osd_fsid, 'lvm')
  File "/usr/lib/python3.6/site-packages/ceph_volume/systemd/systemctl.py", line 82, in enable_volume
    return enable(volume_unit % (device_type, id_, fsid))
  File "/usr/lib/python3.6/site-packages/ceph_volume/systemd/systemctl.py", line 22, in enable
    process.run(['systemctl', 'enable', unit])
  File "/usr/lib/python3.6/site-packages/ceph_volume/process.py", line 153, in run
    raise RuntimeError(msg)
RuntimeError: command returned non-zero exit status: 1

Thanks for your support

**Comment by osaffer** [comment]
Ok, the correct command seems to be : 

 ceph-volume lvm activate --no-systemd --all
--> Activating OSD ID 8 FSID ab18a5e2-3594-4b56-b369-b5bd9a87f4b1
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-8
Running command: /usr/bin/ceph-bluestore-tool --cluster=ceph prime-osd-dir --dev /dev/ceph-4a1fc2a9-5b89-4a3b-bdc7-723013003786/osd-data-8ed2318f-d7db-4d1a-b38f-4f9a3b4c9820 --path /var/lib/ceph/osd/ceph-8 --no-mon-config
Running command: /usr/bin/ln -snf /dev/ceph-4a1fc2a9-5b89-4a3b-bdc7-723013003786/osd-data-8ed2318f-d7db-4d1a-b38f-4f9a3b4c9820 /var/lib/ceph/osd/ceph-8/block
Running command: /usr/bin/chown -h ceph:ceph /var/lib/ceph/osd/ceph-8/block
Running command: /usr/bin/chown -R ceph:ceph /dev/dm-0
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-8
--> ceph-volume lvm activate successful for osd ID: 8
--> Activating OSD ID 4 FSID ba47c6d0-7741-4df4-a98b-05125c21a440
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-4
Running command: /usr/bin/ceph-bluestore-tool --cluster=ceph prime-osd-dir --dev /dev/ceph-6583645f-a2ec-4a02-97a0-838638bd695a/osd-data-02721f33-f2f1-41af-b648-5faf7962a7b9 --path /var/lib/ceph/osd/ceph-4 --no-mon-config
Running command: /usr/bin/ln -snf /dev/ceph-6583645f-a2ec-4a02-97a0-838638bd695a/osd-data-02721f33-f2f1-41af-b648-5faf7962a7b9 /var/lib/ceph/osd/ceph-4/block
Running command: /usr/bin/chown -h ceph:ceph /var/lib/ceph/osd/ceph-4/block
Running command: /usr/bin/chown -R ceph:ceph /dev/dm-1
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-4
--> ceph-volume lvm activate successful for osd ID: 4
--> Activating OSD ID 1 FSID 0097e858-8b52-4ed0-b45d-ac29b4396893
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-1
Running command: /usr/bin/ceph-bluestore-tool --cluster=ceph prime-osd-dir --dev /dev/ceph-8ef19d80-240a-48d1-8eae-c49f6997e65e/osd-data-da4ef665-32dd-4e40-8ab1-ecee98e4d6ea --path /var/lib/ceph/osd/ceph-1 --no-mon-config
Running command: /usr/bin/ln -snf /dev/ceph-8ef19d80-240a-48d1-8eae-c49f6997e65e/osd-data-da4ef665-32dd-4e40-8ab1-ecee98e4d6ea /var/lib/ceph/osd/ceph-1/block
Running command: /usr/bin/chown -h ceph:ceph /var/lib/ceph/osd/ceph-1/block
Running command: /usr/bin/chown -R ceph:ceph /dev/dm-2
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-1
--> ceph-volume lvm activate successful for osd ID: 1



But I cannot see them up when I do a ceph osd tree.

The keyring file is missing in /etc/ceph/admin-keyring-store

....

**Comment by travisn** [comment]
How are you starting the OSDs manually? Are you running those commands somewhere other than the OSD pods? Are the OSD pods not in running state? Or they are just crash looping? If you look on the host, do you see the LVs for the ceph osd with `lsblk`?

**Comment by osaffer** [comment]
All my pods are running, I am just trying to make osd up.

Otherwise I ran these commands inside the pod. 
 I even dont know if I lost my data or not ...

**Comment by osaffer** [comment]
43:480  0         0 disk 
root@adakub2:~# lsblk 
NAME                                                                                                 MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda                                                                                                    8:0    0    25G  0 disk 
├─sda1                                                                                                 8:1    0 478.5M  0 part /boot
├─sda2                                                                                                 8:2    0     1K  0 part 
└─sda5                                                                                                 8:5    0  24.5G  0 part 
  └─vg1-root                                                                                         253:3    0  24.5G  0 lvm  /
sdb                                                                                                    8:16   0     2G  0 disk 
sdc                                                                                                    8:32   0    20G  0 disk 
└─ceph--1a5f85a9--7359--43d9--953a--81f54dedd3be-osd--data--d510b04a--d926--44f1--ad03--b9b57f521399 253:2    0    20G  0 lvm  
sdd                                                                                                    8:48   0    10G  0 disk 
└─ceph--66a68e8b--2673--44a7--99e2--d6a0b053b0ff-osd--data--c084025a--fc0d--4f6e--bb47--0640290af610 253:1    0    10G  0 lvm  
sde                                                                                                    8:64   0    20G  0 disk 
└─ceph--0c3476af--5cb0--4103--a041--0216210761c1-osd--data--6287f955--2cf4--48fa--b044--dcac57a59df0 253:0    0    20G  0 lvm  
sr0                                                                                                   11:0    1  1024M  0 rom  

Yes for all nodes
